src/pyvesync/vesync.py

```diff
@@ -455,6 +526,16 @@ async def async_call_api(
             LibraryLogger.log_api_exception(logger, exception=e, request_body=req_dict)
             raise
 
+    def _api_base_url_for_current_region(self) -> str:
```

Comment:
Why not make this have the same logic as the app, ie: 

```
if self.region in ['US', 'CA', 'MX', 'JP']:
    return 'https://smartapi.vesync.com'
return 'https://smartapi.vesync.eu'
```

Reply:
From what I could tell, in the app they have `com.vesync.config.access.model.AccessUrlConfig` which defines a list with all URLs and their country/region codes.
Then, in `hl0.c.h()`, they iterate over that list and find the matching one, defaulting to `US`.

I thought it would make sense to default to the global URL as well, but we could improve readability by clearly defining this mapping. This was just the quickest way I could get it to work.